relay-filter/src/config.rs

```diff
@@ -192,6 +196,102 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: Option<RuleCondition>,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or no condition was supplied.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || self.condition.is_none()
+    }
+}
+
+/// Configuration for a set of ordered filters.
+#[derive(Clone, Debug, Default)]
+pub struct OrderedFilters(pub Vec<(String, GenericFilterConfig)>);
```

Comment:
Do we need these structs outside `relay-filter`? Same with `GenericFilterConfig` above.
```suggestion
pub(crate) struct OrderedFilters(pub Vec<(String, GenericFilterConfig)>);
```